core/iwasm/aot/aot_runtime.c

```diff
@@ -2535,6 +2535,64 @@ aot_table_init(AOTModuleInstance *module_inst, uint32 tbl_idx,
                 length * sizeof(table_elem_type_t));
 }
 
+#if WASM_ENABLE_GC != 0
+void
+aot_gc_table_init(AOTModuleInstance *module_inst, uint32 tbl_idx,
```

Comment:
I was thinking of a corner case:

The enable_gc is disabled in wamrc, it's only generates an old version aot file containing table.init opcode; but the WASM_ENALBE_GC is enabled in iwasm, when running `iwasm test.aot`, it calls the new version of aot_table_init(since macro is enabled), it will cause a problem.

So I used a different function to distinguish which function needs to be called.